### ISSUE TITLE
After Fedora system upgrade, cockpit certificate shown as "revoked"

### ISSUE THREAD

**Issue by c3d** [cockpit-project/cockpit#13585]
Cockpit version: cockpit-212-1.fc32.x86_64
OS: Fedora 32 (Rawhide)
Page: 

Connecting to "machine:9090"

Chrome browser indicates that the certificate is revoked.
Safari browser indicates that the certificate is "not standard compliant"

Steps to reproduce

1. (After a system upgrade from an older release, I think it was Fedora 30)
2. Connect to "machine:9090" using a web browser

The certificate reported by the browser is stored at `/etc/cockpit/ws-certs.d/0-self-signed.cert` and was apparently created during the upgrade (date is roughly the date of the upgrade).

Chrome's report is:

Your connection is not private
Attackers might be trying to steal your information from big (for example, passwords, messages or credit cards). Learn more
NET::ERR_CERT_REVOKED
Subject: big

Issuer: big

Expires on: 8 Apr 2030

Current date: 20 Feb 2020

PEM encoded chain:
-----BEGIN CERTIFICATE-----
MIIEozCCAougAwIBAgIIIMbfOM9ibS8wDQYJKoZIhvcNAQELBQAwZzELMAkGA1UE
[...]
riYbCpZGI0K2bb4nafXdrPDYYkOYZBTIr1VLouIXJR3tLVdKp4gd
-----END CERTIFICATE-----

Safari shows that the certificate is "not valid before Wednesday 19 February 2020 at 16:24:15 Central European Standard Time", and that the certificate's name is "big", which is also the machine's name.

If I allow Safari to proceed (Chrome won't let me), then cockpit seems to operate normally.

**Comment by martinpitt** [comment]
Could you please run

    sudo openssl x509 -in /etc/cockpit/ws-certs.d/0-self-signed.cert -text

and copy the output here? (The text part is enough, no need for the `BEGIN CERTIFICATE...` binary blob). If the cert was really generated on upgrade (weird, why would it be?), there may be some funky time zone difference somewhere -- but today is definitively later than Feb 19.

**Comment by c3d** [comment]
@martinpitt Here is the output:

Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 2361820489692900655 (0x20c6df38cf626d2f)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = US, O = 2aeafeafc54d4aef9dddbf70379e61b0, OU = ca-4651378047968192868, CN = big
        Validity
            Not Before: Feb 19 15:24:15 2020 GMT
            Not After : Apr  8 08:04:15 2030 GMT
        Subject: C = US, O = 2aeafeafc54d4aef9dddbf70379e61b0, CN = big
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:a8:bd:ac:96:f2:35:c0:c8:87:0a:ca:3b:ea:e0:
                    36:c8:cf:89:ea:bb:93:54:7a:dc:23:a7:4f:75:f0:
                    e7:cd:43:8b:40:d1:5c:a1:ec:c8:6a:4c:bb:22:b5:
                    1f:7d:99:eb:b9:02:b2:2d:c4:d8:42:be:ba:3e:17:
                    9e:27:1c:aa:55:1d:3a:04:bf:e9:d3:3b:6b:a4:35:
                    c7:28:6d:db:7e:88:3e:12:6c:71:dd:1a:a3:69:5e:
                    5c:4f:e8:a0:55:46:67:19:23:9d:05:bf:20:9b:2e:
                    78:ff:89:8a:62:09:34:e1:54:69:0a:81:c0:e0:a2:
                    23:f1:22:e4:2d:26:97:66:82:91:3c:b3:7a:e3:68:
                    c6:69:70:a3:4f:c5:53:2d:b0:8b:a1:88:8f:49:9d:
                    c8:32:b7:ca:05:0a:a1:7e:0c:7a:74:62:95:a7:e7:
                    03:b9:82:0a:4d:50:22:49:e1:8a:87:a2:ba:d4:e6:
                    91:d6:65:f4:2c:cb:ca:6b:8f:3d:37:71:f9:50:68:
                    4c:3e:08:3f:20:13:5f:76:23:5f:ee:c7:95:47:05:
                    3d:a9:6d:aa:ce:91:d1:98:47:93:ba:aa:64:1e:b9:
                    93:7a:da:9d:d8:d7:38:4e:de:e3:a1:80:6d:58:7a:
                    08:52:51:db:90:9c:66:e1:ef:ba:60:b6:78:ac:73:
                    3a:ef
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Subject Alternative Name: 
                DNS:big, DNS:localhost, IP Address:127.0.0.1
            X509v3 Authority Key Identifier: 
                keyid:FF:DA:04:F1:27:92:6B:AC:88:C5:33:89:40:C3:D1:2E:1C:EA:86:93

    Signature Algorithm: sha256WithRSAEncryption
         33:91:83:8b:f3:23:95:bf:4c:94:ae:0b:39:89:65:56:72:5d:
         d7:ac:b4:64:83:80:16:d5:ac:cb:39:50:c5:1a:d7:c7:72:07:
         9a:b1:69:dc:cc:a5:13:d0:1c:75:c0:01:a7:c0:43:3d:db:91:
         92:9c:30:20:cd:97:9e:f0:a7:13:54:17:a3:45:1f:fb:76:d5:
         55:90:21:f9:a2:7e:d6:45:ac:86:df:64:9b:05:f1:ee:f0:76:
         bb:de:24:f2:2d:af:a1:39:39:1b:38:82:35:73:31:5e:7b:79:
         a3:ae:ca:af:de:25:a5:6d:aa:db:5f:fd:32:4a:95:ee:5c:23:
         21:de:8f:89:f8:71:36:93:2c:c6:8a:fc:98:9a:42:eb:fe:25:
         db:d1:d5:c8:7f:e0:e4:f3:4f:33:eb:3e:eb:11:8d:3c:22:50:
         e4:ca:d3:52:4e:b2:fe:a6:b3:47:6c:22:ec:7c:0f:ea:94:68:
         73:0f:9c:77:2b:bc:5f:20:0e:0f:e2:d9:a7:69:e9:8e:3a:d0:
         ba:79:c6:89:dc:d1:23:98:8c:3e:f8:63:56:ff:3e:32:68:51:
         07:19:e2:25:9c:9d:cd:4a:48:f9:b8:ed:db:b8:e3:e9:fa:35:
         87:38:a3:4a:56:ab:00:3b:74:f8:d3:32:3c:d1:e2:a4:8e:f9:
         f8:f5:f4:18:a8:27:07:60:0f:1e:8a:ae:23:ce:90:eb:bd:77:
         32:08:6e:8a:83:94:c8:c4:6c:54:c5:71:65:62:42:f6:43:65:
         26:9a:6f:05:d8:b7:a7:c5:ca:a9:02:e6:30:bf:a5:bb:f6:54:
         aa:20:ea:74:da:ef:65:29:44:36:e0:29:06:fc:55:0d:1f:34:
         d0:08:fa:f5:a9:b3:0b:51:43:e2:dc:02:41:95:ec:69:a9:92:
         e7:17:bd:c9:65:03:38:17:95:9a:3c:bd:2d:1c:b7:82:19:bf:
         4f:34:1d:3d:f7:d7:1e:de:45:be:7b:fd:7a:f8:64:81:f6:ec:
         28:e0:bd:8b:67:2a:d4:4c:28:e8:ee:c0:58:c4:74:ef:37:55:
         52:a0:0d:f9:af:51:a4:e1:25:c8:bc:47:24:1b:d2:a0:f8:89:
         77:89:b0:c9:e8:1c:8d:12:52:e1:c9:07:24:ed:41:e9:15:29:
         86:b0:a7:5e:d2:d6:ad:6e:95:c0:46:92:87:a9:cf:dc:c6:f6:
         92:fc:2e:b0:a2:14:81:60:93:bd:2c:c2:0c:65:51:3c:9a:be:
         e1:a9:45:d2:53:ae:26:1b:0a:96:46:23:42:b6:6d:be:27:69:
         f5:dd:ac:f0:d8:62:43:98:64:14:c8:af:55:4b:a2:e2:17:25:
         1d:ed:2d:57:4a:a7:88:1d

**Comment by martinpitt** [comment]
Thanks. That certificate looks fine (at least the dates), I was hoping for some funky dates, due to the ""not valid before Wednesday 19 February 2020". Is it still broken now (two days after the report)?

**Comment by c3d** [comment]
It still looks broken, but I'm not exactly in the same conditions, because I'm not at home at the moment, so I had to open an ssh tunnel to test. Here is what I did:

```
ssh -L 3030:target:9090 gateway
```

and then connected locally on port 3030. Initially, it seemed to work, yay, but then when I tried to check the certificate being used, I noticed that it had selected `http://` (duh). Trying to use `https://localhost:3030` showed the same certificate error.

Notice that the way this is reported by Chrome is
```
NET::ERR_CERT_REVOKED
```

I think that's not what you get if the certificate has expired. (I believe you get a `_EXPIRED` in that case).
